docs/preview/02-Features/02-message-handling/01-service-bus.md

```diff
@@ -140,111 +131,105 @@ public class Startup
 {
     public void ConfigureServices(IServiceCollection services)
     {
-        // Specify the name of the Service Bus Queue:
-        services.AddServiceBusQueueMessagePump(
-            "My-Service-Bus-Queue-Name",
-            "ARCUS_SERVICEBUS_ORDERS_CONNECTIONSTRING");
+        services.WithServiceBusMessageHandler<OrderMessageHandler, Order>(context => context.Properties["MessageType"].ToString() == "Order");
+    }
+}
+```
 
-        // Specify the name of the Service Bus Topic, and provide a name for the Topic subscription:
-        services.AddServiceBusMessageTopicMessagePump<OrdersMessageHandler>(
-            "My-Service-Bus-Topic-Name",
-            "My-Service-Bus-Topic-Subscription-Name",
-            "ARCUS_SERVICEBUS_ORDERS_CONNECTIONSTRING");
+> Note that the order in which the message handlers are registered is important in the message processing.
+> In the example, when a message handler above this one is registered that could also handle the message (same message type) than that handler may be chosen instead of the one with the specific filter.
```

Comment:
may be chosen or will be chosen ?

The first option implies that you're not certain which handler will be used at runtime.  The latter means that you're absolutely sure that the first registered handler will be used.

I'd extend line 139 with an additional sentence that says that the first registered message-handler that is eligeable to process a certain message, will be used.

Reply:
> may be chosen or will be chosen ?

> The first option implies that you're not certain which handler will be used at runtime. The latter means that you're absolutely sure that the first registered handler will be used.

It's still 'may' here, I think, as the message type is not solely responsible for the message routing. One can add additional message type filters, message body filters, message context filters and even bring in their own serialization for the same message type.